core/src/main/scala/spinal/core/sim/SpinalSpec.scala

```diff
@@ -0,0 +1,273 @@
+package spinal.core.sim
+
+import org.scalatest.freespec.AnyFreeSpec
+import spinal.core.{Component, Nameable}
+
+/** Define unit / regression tests for a component */
+abstract class SpinalSpec[Dut <: Component] extends AnyFreeSpec with spinal.idslplugin.ValCallback {
```

Comment:
Good catch! But now I have to provide 2 type parameters on Protocol (see test file)

Reply:
I fixed it; the 2 type parameters were `Dut` (`HardDut`, the actual `Component` to compile) and `SoftDut` (possibly abstracted what tests are run on, the only thing used in the interface).

So I made `Bench` abstract, moved `Dut` in an internal type so that the only type which remains in the type signature is `SoftDut`, making it possible to have only one type parameter for `Protocol`.